apps/guardian-ui/src/setup/FederationSetup.tsx

```diff
@@ -82,6 +83,22 @@ export const FederationSetup: React.FC = () => {
       });
   }, [api, dispatch]);
 
+  useEffect(() => {
```

Comment:
nit: We can early return if `progress !== SetupProgress.Start` to avoid checking the query params.
lint: Add `progress` to the params of this useEffect?